src/twisted/web/client.py

```diff
@@ -1717,6 +1726,7 @@ def request(self, method, uri, headers=None, bodyProducer=None):
 
         @see: L{twisted.web.iweb.IAgent.request}
         """
+        uri = _ensureValidURI(uri.strip())
```

Comment:
Should the `.strip()` go in `_ensureValidURI` ?

Reply:
I don't think so, because not all APIs that end up needing this strip their URI.